tests/testthat/test-methods.R

```diff
@@ -144,7 +144,11 @@ test_that("split.lint works as intended", {
 test_that("within.list is dispatched", {
   l <- lint(text = "a=1\nb=2", linters = infix_spaces_linter())
   expect_silent({
-    l <- lapply(l, within, line_number <- line_number + 1L)
+    l <- lapply(l, function(li) {
```

Comment:
Seems there's a fundamental conflict among linters here. The current code violates `unnecessary_lambda_linter()`...

I'll proceed by `# nolint`ing the original code. It's much more concise anyway.

(actually, since we haven't turned on `implicit_assignment_linter()`, I'll just revert this change for now)